File: Java/Platform/Basics/ThreadTest/Program3.java
class Program {
	
	private static ThreadLocal<String> client = new ThreadLocal<>();

	private static void handleJob(int id) {
		System.out.printf("Thread<%x> has accepted job<%d> for %s%n", Thread.currentThread().hashCode(), id, client.get());
		Worker.doWork(10 * id);
		System.out.printf("Thread<%x> has finished job<%d> for %s%n", Thread.currentThread().hashCode(), id, client.get());
	}

	public static void main(String[] args) {
		int n = args.length > 0 ? Integer.parseInt(args[0]) : 1;
		Thread child = new Thread(() -> {
			client.set("Jack");
			handleJob(n);
		});
		child.setDaemon(n > 10);
		child.start();
		client.set("Jill");
		handleJob(2);
	}
}

File: dotnet/Platform/Basics/ThreadTest/DemoApp/Program3.cs
class Program
{
    [ThreadStatic]
    private static string client;

    private static void HandleJob(int id)
    {
        Console.WriteLine("Thread<{0}> has accepted job<{1}> for client<{2}>", Thread.CurrentThread.ManagedThreadId, id, client);
        Worker.DoWork(10 * id);
        Console.WriteLine("Thread<{0}> has finished job<{1}> for client<{2}>", Thread.CurrentThread.ManagedThreadId, id, client);
    }   

    static void Main(string[] args)
    {
        int n = args.Length > 0 ? int.Parse(args[0]) : 1;
        Thread child = new Thread( () =>
        {
            client = "Jack";
            HandleJob(n);
        });
        child.IsBackground = n > 10;
        child.Start();
        client = "Jill";
        HandleJob(2);
    }
}
